# 57) Tek Örneklem Z Testi
Tek Örneklem Z Testi, bir örneklemin ortalamasının, **belirli/iddia edilen bir değere** eşit olup olmadığını test eder. "Popülasyon ortalaması X'tir" iddiasını (H0), elimizdeki tek bir örneklemle sınarız.

## Ne Zaman Kullanılır?
- Popülasyonun standart sapması ($\sigma$) **biliniyorsa**, VEYA
- Örneklem büyüklüğü **n ≥ 30** ise (Merkezi Limit Teoremi sayesinde)

Bu koşullar sağlanmıyorsa (n<30 ve $\sigma$ bilinmiyorsa), bir sonraki konuda göreceğimiz **tek örneklem t-testi** kullanılır.

## Hipotezler
- **H0:** $\mu = \mu_0$ (popülasyon ortalaması, iddia edilen değere eşittir)
- **H1:** $\mu \ne \mu_0$ (çift kuyruklu) veya $\mu > \mu_0$ / $\mu < \mu_0$ (tek kuyruklu)

## Test İstatistiği Formülü
$$Z = \frac{\bar{x} - \mu_0}{\sigma / \sqrt{n}}$$

Bu formül tanıdık — Konu 54'te (Test İstatistiği) gördüğümüz genel iskeletin ($\frac{\text{Bulunan} - \text{Varsayılan}}{\text{Standart Hata}}$) birebir uygulaması. Payda, Konu 19'daki Standart Hata formülü.

## Karar Süreci (Özet Akış)
1. H0 ve H1'i kurup, α'yı belirleyelim (genelde 0.05).
2. Z test istatistiğini hesaplıyoruz.
3. Z'yi kullanarak p-değerini buluyoruz (`stats.norm.cdf()` ile).
4. p'yi α ile karşılaştırıp, karar veriyoruz.

## Örnek Senaryo
Bir fabrika, ürettiği paketlerin ortalama ağırlığının **500 gram** olduğunu iddia ediyor, ve geçmiş verilerden popülasyon std'sinin 
$\sigma=15$ gram olduğu biliniyor. 40 paketlik bir örneklem alındı, ortalama 495 gram çıktı. Paketlerin gerçekten iddia edilenden hafif olup olmadığını test edelim.

In [1]:
import numpy as np
from scipy import stats
from math import sqrt
# H0: μ = 500 (paketlerin gerçek ortalama ağırlığı 500 gramdır, iddia doğrudur)
# H1: μ < 500 (paketler iddia edilenden daha hafiftir) tek kuyruklu test
x_ort = 495
populasyon_ort = 500
std = 15
n = 40
z = (x_ort - populasyon_ort) / (std / (sqrt(n)))
print(f"Z istatistiği: {z}")
# p-değerini bulalım — tek kuyruklu (sol kuyruk, çünkü H1: μ < 500)
p_degeri = stats.norm.cdf(z)
print(f"p-değeri: {p_degeri}")

alpha = 0.05
if p_degeri < alpha:
    print(f"p ({p_degeri:.4f}) < α ({alpha}) → H0 reddedilir. Paketler istatistiksel olarak anlamlı şekilde daha hafiftir.")
else:
    print(f"p ({p_degeri:.4f}) >= α ({alpha}) → H0 reddedilemedi.")

Z istatistiği: -2.1081851067789197
p-değeri: 0.017507490509831244
p (0.0175) < α (0.05) → H0 reddedilir. Paketler istatistiksel olarak anlamlı şekilde daha hafiftir.


### Sonuç
Bu doğrultuda fabrikanın, paketlerin neden iddia edilen ağırlığın altında kaldığını araştırması gerekir. Üretim sürecinde bir sapma mı var, tartı ekipmanlarında bir kalibrasyon sorunu mu var, yoksa hammadde miktarında bir eksiklik mi var, bu noktaların ayrıca incelenmesi 
gerekmektedir.


### Örnek 2
Bir kahve zinciri, müşterilerin ortalama bekleme süresinin 4 dakika olduğunu iddia ediyor. Popülasyon standart sapmasının 𝜎=1.2 dakika olduğu biliniyor. Biz, gerçek bekleme süresinin iddia edilenden farklı (uzun ya da kısa olabilir, yönden emin değilsin) olabileceğini düşünüyoruz. 50 müşterilik bir örneklem topladık, ortalama bekleme süresi 4.3 dakika çıktı.

Görevimiz:
1. H0 ve H1'i kurmak (bu sefer kuyruk yapısına dikkat etmeliyiz — yön belirtiyor muyuz, belirtmiyor muyuz?)
2. Z istatistiğini hesaplamak
3. p-değerini bulmak (ipucu: çift kuyruklu olduğu için, bulduğun tek taraflı p'yi 2 ile çarpmamız gerekecek — çünkü iki kuyruğu da hesaba katmalıyız)
4. α=0.05 ile karşılaştırıp karar vermek

In [5]:
# H0: müşterilerin ortalama bekleme süresi 4 dakikadır (μ = 4).
# H1: müşterilerin ortalama bekleme süresi 4 dakikadan farklıdır (μ ≠ 4).
pop_ort = 4
orneklem_ort = 4.3
sapma = 1.2
boyut = 50
z_istatistigi = (orneklem_ort - pop_ort) / (sapma / sqrt(boyut))
print(f"Z istatistiği: {z_istatistigi}")
z_ust_sinir = stats.norm.ppf(0.975)
z_alt_sinir = stats.norm.ppf(0.025)

if (z_istatistigi < z_alt_sinir) or (z_istatistigi > z_ust_sinir):
    print(f'H0 reddediliyor, müşterilerin ortalama süresi 4 dakikadan anlamlı olarak farklıdır.')
else:
    print(f'H0 reddedilemez, ortalama sürenin 4 dakikadan farklı olduğuna dair yeterli kanıt yoktur.')

p_degeri = 2 * stats.norm.cdf(-abs(z_istatistigi))
print(f"p-değeri: {p_degeri}")

if p_degeri < alpha:
    print("H0 reddedilir")
else:
    print("H0 reddedilemez")

p_degeri_2 = (1 - (stats.norm.cdf(abs(z_istatistigi)))) * 2
print(f"2. p-değeri: {p_degeri_2}")
if p_degeri_2 < alpha:
    print("H0 reddedilir")
else:
    print("H0 reddedilemez")

Z istatistiği: 1.767766952966368
H0 reddedilemez, ortalama sürenin 4 dakikadan farklı olduğuna dair yeterli kanıt yoktur.
p-değeri: 0.0770998717435419
H0 reddedilemez
2. p-değeri: 0.07709987174354183
H0 reddedilemez


### Sonuç
Bir tek kuyruklu (fabrika) ve bir çift kuyruklu (kahve) hipotez testini örnek olarak çözdük. Çift kuyruklu testte p-değerini iki farklı yoldan hesapladık: ilk yöntemde Z'nin mutlak değerini negatife çevirip sol kuyruktaki alanı bulduk ve 2 ile çarptık; ikinci yöntemde ise Z'nin mutlak değerinin solundaki alanı bulup 1'den çıkararak sağ kuyruktaki alanı bulduk, yine 2 ile çarptık. Normal dağılımın simetrisi sayesinde, her iki yöntem de birebir aynı p-değerine (0.077) ulaştı.